### PR TITLE
Add bucketing and sorting info to HiveConnectorSplit

### PR DESCRIPTION
**Pull request by oerling**

### CODE REVIEW

File: velox/connectors/hive/HiveConnectorSplit.h [hunk 1]
```diff
@@ -22,14 +22,28 @@
...
+
 struct HiveConnectorSplit : public connector::ConnectorSplit {
   const std::string filePath;
   dwio::common::FileFormat fileFormat;
   const uint64_t start;
   const uint64_t length;
   const std::unordered_map<std::string, std::optional<std::string>>
       partitionKeys;
-  std::optional<int32_t> tableBucketNumber;
+  const std::optional<int32_t> tableBucketNumber;
```
Comment (mbasmanova) [inline velox/connectors/hive/HiveConnectorSplit.h:42]: These are table-level properties, hence, they belong to HiveTableHandle, not HiveConnectorSplit.

**Comment by oerling** [comment]
The values in question do not occur in the table handles but they do occur in the split. They do occur in the write table handle but we need these values for read.

**Comment by mbasmanova** [comment]
> The values in question do not occur in the table handles but they do occur in the split. They do occur in the write table handle but we need these values for read.

This is so strange. Is this because different partitions of the same table can be bucketed/sorted differently? It would be nice to add a comment explaining this as I'm sure other readers will have similar questions.

There is a lot of "optional" parameters now in the constructor, but these new parameters have to all be set or unset together as a unit. They cannot have independent "set" status, right? Would it make sense to add a separate struct to hold these?

**Comment by oerling** [comment]
It would of course be good if the plan node knew details about the table it is about to process. So this makes no sense. True, by minimum need to know, the reader must not know about these things. But then the split should also not know about these.   

 

So for now the plan adaptation must work based on the split and not the plan node.

 

Different bucketing in different partitions would not work in the plans that we see.

**Comment by mbasmanova** [comment]
Let's dig in a bit more then to understand why Presto puts this information in the split. It may be cleaner to fix Presto, then propagate the hack into Velox.

**Comment by mbasmanova** [comment]
Closing stale PRs. Feel free to re-open if decide to continue working on this.

**Pull request closed without merging**